src/Views/OperatingSystemView.vala

```diff
@@ -547,9 +565,17 @@ public class About.OperatingSystemView : Gtk.Box {
                 }
                 break;
             case DOWNLOADING:
+                update_progress_revealer.reveal_child = current_state.percentage > 0;
+                update_progress_bar.value = current_state.percentage;
+
+                download_size_remaining = current_state.download_size_remaining;
+                if (download_size_remaining > download_size_max) {
+                    download_size_max = download_size_remaining;
+                }
+
                 updates_image.icon_name = "browser-download";
                 updates_title.label = _("Downloading Updates");
-                updates_description.label = current_state.message;
+                updates_description.label = "%s %s".printf (current_state.message, get_progress_text ());
```

Comment:
Probably should make sure to use tabular numbers so the label doesn't bounce around as values change:

`<span font-features='tnum'>%s</span>`